File: ucrop/src/main/java/com/yalantis/ucrop/adapter/SizeAdapter.java
package com.yalantis.ucrop.adapter;

import android.content.Context;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.yalantis.ucrop.R;
import com.yalantis.ucrop.model.AspectRatio;
import com.yalantis.ucrop.view.widget.AspectRatioImageView;

import java.util.ArrayList;

public class SizeAdapter extends RecyclerView.Adapter<SizeAdapter.ItemHolder> {

    private ArrayList<AspectRatio> list;

    public SizeAdapter(Context context) {
        list = new ArrayList<>();
        list.add(new AspectRatio("1:1", 1, 1));
        list.add(new AspectRatio("3:4", 3, 4));
        list.add(new AspectRatio("3:2", 3, 2));
        list.add(new AspectRatio("16:9", 16, 9));
    }

    @NonNull
    @Override
    public ItemHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ItemHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.child_crop_size, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ItemHolder holder, int position) {
        holder.imageView.setXYSize(list.get(position).getAspectRatioX() / list.get(position).getAspectRatioY());
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    class ItemHolder extends RecyclerView.ViewHolder {

        final AspectRatioImageView imageView;

        public ItemHolder(@NonNull View itemView) {
            super(itemView);

            imageView = itemView.findViewById(R.id.imageViewRatioChild);

        }
    }
}

File: ucrop/src/main/java/com/yalantis/ucrop/adapter/SizeAdapter.kt
package com.yalantis.ucrop.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.yalantis.ucrop.R
import com.yalantis.ucrop.model.AspectRatio
import kotlinx.android.synthetic.main.child_crop_size.view.*
import java.util.*

class SizeAdapter : RecyclerView.Adapter<ItemHolder>() {
    private val list: ArrayList<AspectRatio> = ArrayList()
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemHolder {
        return ItemHolder(LayoutInflater.from(parent.context).inflate(R.layout.child_crop_size, parent, false))
    }

    override fun onBindViewHolder(holder: ItemHolder, position: Int) {
        holder.itemView.imageViewRatioChild.setXYSize(list[position].aspectRatioX / list[position].aspectRatioY);
    }

    override fun getItemCount(): Int {
        return list.size
    }

    init {
        list.add(AspectRatio("1:1", 1f, 1f))
        list.add(AspectRatio("3:4", 3f, 4f))
        list.add(AspectRatio("3:2", 3f, 2f))
        list.add(AspectRatio("16:9", 16f, 9f))
    }
}